In [1]:
%load_ext autoreload
%autoreload 2

In [42]:
import pandas as pd
import torch

from src.config import CONFIG
from src.dataset import SBICDataset
from src.train_utils import make_tokinzer, make_model, make_dataloader, evaluate
from src.utils import print_evaluation_results

In [3]:
tokenizer = make_tokinzer(CONFIG.hp)
model = make_model(CONFIG.hp, tokenizer)
checkpoints = "distilgpt2_full_1"
model.load_state_dict(torch.load(f"checkpoints/{checkpoints}.pt"))

List of all special token and its token_id:
 - ['<|endoftext|>', '<|sep|>', '<|pad|>', '<|offY|>', '<|offN|>', '<|sexY|>', '<|sexN|>', '<|intY|>', '<|intN|>', '<|grpY|>', '<|grpN|>', '<|ingrpN|>', '<|ingrpY|>']
 - [[50256], [50258], [50257], [50259], [50260], [50261], [50262], [50263], [50264], [50265], [50266], [50267], [50268]]
Model vocab resize: 50269
Model eos token: 50256
Model pad token: 50257
Model sep token: 50258


<All keys matched successfully>

In [44]:
split = "validation"
data = pd.read_pickle(CONFIG.dataset.val_data_agg).to_numpy()
dataset = SBICDataset(data, tokenizer, is_training=False)
dataloader = make_dataloader(dataset, model, tokenizer, CONFIG.hp, split="validation")

val_res = evaluate(model, tokenizer, dataloader, CONFIG.hp)

print_evaluation_results(split=split, res=val_res)

Classification F1 on validation set: avg=0.697
 - Offensive: 0.652
 - Intent: 0.609
 - Sex: 0.840
 - Group: 0.479
 - In-Group: 0.905
Minority RougeL-f1 on validation set: 0.711
Stereotype RougeL-f1 on validation set: 0.282


                                                 
Classification F1 on validation set: avg=0.697
 - Offensive: 0.652
 - Intent: 0.609
 - Sex: 0.840
 - Group: 0.479
 - In-Group: 0.905
 
Minority RougeL-f1 on validation set: 0.813 <br>
Stereotype RougeL-f1 on validation set: 0.508 <br>

In [43]:
split = "test"
data = pd.read_pickle(CONFIG.dataset.test_data_agg).to_numpy()
dataset = SBICDataset(data, tokenizer, is_training=False)
dataloader = make_dataloader(dataset, model, tokenizer, CONFIG.hp, split="validation")

test_res = evaluate(model, tokenizer, dataloader, CONFIG.hp)

print_evaluation_results(split=split, res=test_res)

Classification F1 on test set: avg=0.704
 - Offensive: 0.654
 - Intent: 0.617
 - Sex: 0.832
 - Group: 0.508
 - In-Group: 0.911
Minority RougeL-f1 on test set: 0.863
Stereotype RougeL-f1 on test set: 0.531
